### DISCUSSION TITLE
Retrying the resource load via ResourceRequestHandler.OnResourceResponse

### DISCUSSION THREAD

**Discussion by tobiigyk** [cefsharp/CefSharp#3752] [General]
Hi!
I've used ResourceRequestHandler.OnResourceResponse to postpone resources loading (and rendering a new page) until my local job is not finished with previous page. And it worked in 73 version, but does not work in 91.

The comments in code for this part remain the same:
    > To redirect or retry the resource load optionally modify <paramref name="request"/> and return true.

To recheck I've added 2 classes in CefSharp.Wpf.Example project.

And used URL check instead of more specific condition (I understand that cancellation of resource loading should be done in OnBeforeResourceLoad, just didn't come up with better example here).

So for this example expected behavior: google page won't be loaded.

Actual: it is loaded.

```
    public sealed class ExtendedResourceRequestHandler : ResourceRequestHandler
    {
        protected override bool OnResourceResponse(
            IWebBrowser chromiumWebBrowser,
            IBrowser browser,
            IFrame frame,
            IRequest request,
            IResponse response)
        {
            if (request.Url.Contains("google"))
            {
                return true;
            }

            return base.OnResourceResponse(chromiumWebBrowser, browser, frame, request, response);
        }
    }    
```

and

```
    public sealed class ExtendedRequestHandler : RequestHandler
    {
        protected override IResourceRequestHandler GetResourceRequestHandler(
            IWebBrowser chromiumWebBrowser,
            IBrowser browser,
            IFrame frame,
            IRequest request,
            bool isNavigation,
            bool isDownload,
            string requestInitiator,
            ref bool disableDefaultHandling)
        {
            return new ExtendedResourceRequestHandler();
        }
    }
```

ExtendedRequestHandler is set to browser, so breakpoints work during debug in both classes.

**Comment by amaitland** [comment]
> And it worked in 73 version, but does not work in 91.

The CEF behaviour has likely changed, you can review the latest API doc at

https://github.com/chromiumembedded/cef/blob/a4d71d8fe2d7422c8e48128f55a503753e2d01c8/include/cef_resource_request_handler.h#L146

> So for this example expected behavior: google page won't be loaded.

I don't believe this is correct,  OnResourceResponse is not intended to cancel a request. 

To better understand what's going on you can review the source.

https://github.com/chromiumembedded/cef/blob/ba46b8c53ed0f46cec82a1aebaec173784deb6e4/libcef/browser/net_service/resource_request_handler_wrapper.cc#L793

**Reply by amaitland** [reply]
> CEF now supports the Chromium Network Service: If you are upgrading from a version prior to 75.1.x please make sure you read #2743.

As per the release notes, major changes starting in version 75
